src/wallet/test/wallet_tests.cpp

```diff
@@ -37,14 +37,14 @@ static void AddKey(CWallet& wallet, const CKey& key)
 BOOST_FIXTURE_TEST_CASE(scan_for_wallet_transactions, TestChain100Setup)
 {
     auto chain = interfaces::MakeChain();
+    LockAnnotation lock(::cs_main);
```

Comment:
I might be missing something but shouldn't `LockAnnotation` be used only when we are holding the mutex but the compiler thread analysis doesn't understand it? In this case we aren't holding `cs_main` here, right?

If it is used simply to opt-out of thread-safety analysis (when we are not holding the mutex) then `NO_THREAD_SAFETY_ANALYSIS` should be used, no?

Reply:
The tests are single threaded, so taking a lock is equivalent to pretending it is taken.

Not sure how I can use `NO_THREAD_SAFETY_ANALYSIS` to achieve that.

Reply:
Yes, I know the tests are single-threaded but I think it might be confusing for newcomers if we use `LockAnnotation` to both a.) guarantee to the compiler that the lock is taken (like in `src/interfaces/chain.cpp`), and b.) to opt-out of thread safety analysis when the lock is not taken (like in this file).

Just to make it crystal clear to readers what we are doing we could use different names for these two use-cases? (If using `NO_THREAD_SAFETY_ANALYSIS` is not an option.)

Something like `struct SCOPED_LOCKABLE DisableThreadSafetyAnalysis` (like `LockAnnotation`) or `#define DISABLE_THREAD_SAFETY_ANALYSIS(cs, code) [&] NO_THREAD_SAFETY_ANALYSIS { code; }()`?

Reply:
Added a comment instead. Is that fine?